app/validators/suffering_validator.rb

```diff
@@ -1,7 +1,11 @@
 class SufferingValidator < ActiveModel::Validator
   def validate(record)
     if !record.diagnosis_date.nil? && record.diagnosis_date > Date.today
-      record.errors[:diagnosis_date] << "Diagnosis date must be in the past"
+      record.errors.add(:diagnosis_date, 'Diagnosis date must be in the past')
+    end
+    if record.dischargement_date.nil? && record.status != 'healty'
```

Comment:
Actually, I think the `dischargement_date` field must be filled for any healthy cat because it indicates when did the cat get recovered of that illness/health condition.